### ISSUE TITLE
`nuitka` CLI not found on Windows if installed with `uv`

### ISSUE THREAD

**Issue by kdeldycke** [Nuitka/Nuitka#3173]
Tl;Dr: On Windows `nuitka` is [reported as an available CLI](https://github.com/kdeldycke/workflows/actions/runs/11533605793/job/32106738267#step:6:1) in a `uv` venv:
```shell-session
$ uv run  
warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
Installed 46 packages in 263ms
Provide a command or script to invoke with `uv run <command>` or `uv run <script>.py`.
The following commands are available in the environment:
- bump-my-version
- distro
- dotenv
- gha-utils
- lizard
- markdown-it
- normalizer
- nuitka
- nuitka-run
- pydoc
- pygmentize
- python
- pythonw
- rich-click
- tabulate
See `uv run --help` for more information.
```

But its invocation fails:
```shell-session
$ uv --no-progress run --frozen -- nuitka --version
  
error: Failed to spawn: `nuitka`
  Caused by: program not found
```

### Context

This [happened in a GitHub action runner on `windows-2022` ](https://github.com/kdeldycke/workflows/actions/runs/11533605793/job/32106738267#step:8:11). But the exact same sequence and invocation [works perfectly on macOS](https://github.com/kdeldycke/workflows/actions/runs/11533403860/job/32106819410) or [on a Linux runner](https://github.com/kdeldycke/workflows/actions/runs/11533403860/job/32106819338#step:7:1).

### Fix

To fix the issue, I have to [replace the invocation by an indirect call](https://github.com/kdeldycke/workflows/actions/runs/11533403860/job/32106819536#step:8:1) to the Python interpreter:
```shell-session
$ uv --no-progress run --frozen -- python -m nuitka --version
```

### Reproduction

To reproduce the issue:

```shell-session
$ git clone https://github.com/kdeldycke/workflows
$ cd workflows
$ python -m pip install uv
$ uv venv
$ source .venv/bin/activate
$ uv pip install .
$ uv pip install -r ./requirements/nuitka.txt
Resolved 3 packages in 723ms
Installed 3 packages in 20ms
 + nuitka==2.4.11
 + ordered-set==4.1.0
 + zstandard==0.23.0
$ uv --no-progress run --frozen -- nuitka --version
```

### Working case

For the record, there is no issue of running the commands above on macOS Sequoia 15.0.1, which outputs the following with the latest Nuitka:
```shell-session
$ uv --no-progress run --frozen -- nuitka --version
2.4.11
Commercial: None
Python: 3.13.0 (main, Oct  7 2024, 05:02:14) [Clang 16.0.0 (clang-1600.0.26.3)]
Flavor: Homebrew Python
GIL: yes
Executable: /Users/kde/workflows/.venv/bin/python3
OS: Darwin
Arch: arm64
macOSRelease: 15.0.1
Version C compiler: /usr/bin/clang (clang 16.0.0).
```

**Comment by kayhayen** [comment]
I did this:

```
python3.10 -m pip install uv
. uv_test/Scripts/activate # your instructions ask for bin here wrongy?
```

I have no `uv` in there, and no `pip` at that point. So am I supposed to install to there from the outside? You might be using uv wrong and getting away with it, I am not sure, I want to git clone uv for a reproducer. Please provide instructions using only uv and pip.

**Comment by kdeldycke** [comment]
I made a simple example at: https://github.com/kdeldycke/nuitka-issue-3173/blob/main/.github/workflows/release.yaml

This is just an empty repository that is trying to reproduce the issue with the simplest step of commands. Which essentially consist in:
```shell-session
$ python -m pip install uv
$ uv venv
$ uv pip install nuitka
$ uv run -- nuitka --version
```

In my example repository I pinned all versions of all dependencies, and also run the same workflow on all platforms. You can then [see that `uv run -- nuitka --version` works everywhere, but not on `windows-2022`](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280568400).

**Comment by kayhayen** [comment]
That is indeed a lot more straightfoward. Not sure, how it could be a Nuitka bug, if they don't install the `scripts` apparently. We don't want to frozen exe that pip creates on Windows, and instead provide cmd files that launch Nuitka with `python -m nuitka` the way it is recommended to be ran.

I think you found an incompatibility of uv there, and should report it. I was able to reproduce it with installing from the git folder, with `python3.10 -m uv pip install ~/repos/Py2C`.

Since uv is supposed to be so fast, I guess they don't have time to copy the scripts specified.

**Comment by kayhayen** [comment]
Somebody needs to open a bug with uv, probably making a small reproducer that uses `setup(scripts=)` for their convenience. I don't think I want to special case uv or anything, but I am open to learning. I am also open to learning that maybe some hacks we do for console scripts are interfering, but I don't think so.

**Comment by kdeldycke** [comment]
Ok, I just reported this issue upstream to `uv`: https://github.com/astral-sh/uv/issues/8770

**Comment by kayhayen** [comment]
They moved it to https://github.com/astral-sh/uv/issues/9151 it seems

**Comment by kdeldycke** [comment]
> `uv run -m nuitka`

Yes, this works, but is just a shorthand for the `uv run python -m nuitka` invokation that has been mentioned as a temporary fix in the description of this issue.

The support for Windows executable extensions is still being discussed upstream in https://github.com/astral-sh/uv/issues/9151

**Comment by kdeldycke** [comment]
I just updated my test case with latest `uv` (0.5.26) and `nuitka` (2.6.2) to confirm the issue is still reproducible: https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/13106670972/job/36562560546

**Comment by kayhayen** [comment]
I was wondering about this too. I am adding support for uv-Python on non-Windows (seems Windows was already working) as part of a Nuitka issue. Unfortunately the static libpython provided is unusable, can somebody create a proper issue for the UV makers to provide the missing static link libries as wel, so it can be used with best performance in Nuitka too?

**Comment by kdeldycke** [comment]
Just to keep track of the progress, the issue can still be reproduced with the `uv 0.6.3` and `nuitka 2.6.7` combo: https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/13517137348

**Comment by kdeldycke** [comment]
Issue has been fixed in uv 0.6.5! 🎉

See:

![Image](https://github.com/user-attachments/assets/55ce0cca-b0a0-4a80-88f5-76fd6f8146b9)

Refs: https://github.com/astral-sh/uv/pull/11888
